data_steward/retraction/create_deid_map.py

```diff
@@ -0,0 +1,180 @@
+CREATE OR REPLACE TABLE `{project}.{dataset}._deid_map` AS (
+SELECT c.person_id, d.person_id AS research_id
+FROM `{project}.{dataset}.observation` c
+JOIN `{project}.{dataset}.observation` d
+ON c.observation_id = d.observation_id
+)
+"""
+
+
+def get_combined_datasets_for_deid_map(project_id):
+    """
+    List all datasets in given project_id, filter out datasets that contain a deid dataset and collect the corresponding
+    combined dataset
+    :param project_id: bq name of project_id
+    :return: list of combined_datasets that should contain a _deid_map table
+    """
+    all_datasets_obj = utils.bq.list_datasets(project_id)
+    all_datasets = [d.dataset_id for d in all_datasets_obj]
+    deid_datasets = []
+
+    # Keep only deid datasets with standard cdr release naming convention
+    release_datasets = [
+        d for d in all_datasets if CURRENT_RELEASE_REGEX.match(d)
+    ]
+    # Filter for deid datasets
+    for dataset in release_datasets:
+        if bool(re.match(DEID_REGEX, dataset)) is True:
+            deid_datasets.append(dataset)
+    # Find corresponding combined dataset for each deid dataset
+    combined_datasets = get_corresponding_combined_dataset(
+        all_datasets, deid_datasets)
+    return combined_datasets
+
+
+def get_corresponding_combined_dataset(all_datasets, deid_datasets):
+    """
+    Takes in all datasets in a project and a list of all the deid datasets in that project and returns the corresponding
+    combined dataset to that deid dataset
+    :param all_datasets: list of all datasets in the given bq project
+    :param deid_datasets: list of all deid datasets in the given bq project
+    :return: list of combine datasets
+    """
+    combined_datasets = []
+    # Get release from deid dataset for both older and current release tags and search for combined dataset
+    for d in deid_datasets:
+        release = d.split('_')[0]
+        older_release = release[1:]
+        combined, older_combined = release + '_combined', older_release + '_combined'
+
+        if combined in all_datasets:
+            combined_datasets.append(combined)
+        if older_combined in all_datasets:
+            combined_datasets.append(older_combined)
+    # Remove duplicates
+    combined_datasets = list(set(combined_datasets))
+    return combined_datasets
+
+
+def check_if_deid_map_exists(project_id, dataset):
+    """
+    Checks if _deid_map table exists in the given dataset.
+    :param project_id: bq name of project_id
+    :param dataset: bq name of dataset
+    :return: returns True, False or 'rename required' which had 'deid_map' table instead of '_deid_map'
+    """
+    table_info_df = utils.bq.get_table_info_for_dataset(project_id, dataset)
+    column_list = table_info_df['table_name'].tolist()
+
+    if 'deid_map' in column_list:
+        return 'rename required'
+    if '_deid_map' in column_list:
+        return True
+    if ['deid_map', '_deid_map'] not in column_list:
+        return False
```

Comment:
I initially used all `if`'s because I wanted it to check all possibilities, which now I'm blanking on why, but with adding if 'deid_map' and '_deid_map' both exist I will need to keep the `if`'s. I will make this adjustment!

Reply:
You could change the order to make this possible: If _deid exists return 'skip', elif deid exists return 'rename', else return 'create'. If both exist, we don't have to worry here since it would be skipped.
Booleans might be confusing since three possibilities exist here. It would help to use constants similar to enum types, like `consts.RENAME = 'rename', consts.SKIP = 'skip', consts.CREATE = 'create'` and returning them for each of these conditions.

Reply:
Ah I see, thank you! I will create a constants file like I see you've done previously with other retraction files and use those for that if statement.